test_conformance/images/kernel_image_methods/test_1D_buffer.cpp

```diff
@@ -0,0 +1,282 @@
+}
+
+int test_get_image_info_1D_buffer(cl_device_id device, cl_context context,
+                                  cl_command_queue queue,
+                                  cl_image_format *format, cl_mem_flags flags)
+{
+    size_t maxWidth;
+    cl_ulong maxAllocSize, memSize;
+    image_descriptor imageInfo = { 0 };
+    RandomSeed seed(gRandomSeed);
+    size_t pixelSize;
+
+    imageInfo.type = CL_MEM_OBJECT_IMAGE1D_BUFFER;
+    imageInfo.format = format;
+    imageInfo.height = imageInfo.depth = imageInfo.slicePitch = 0;
+    pixelSize = get_pixel_size(imageInfo.format);
+
+    int error = clGetDeviceInfo(device, CL_DEVICE_IMAGE_MAX_BUFFER_SIZE,
+                                sizeof(maxWidth), &maxWidth, NULL);
+    error |= clGetDeviceInfo(device, CL_DEVICE_MAX_MEM_ALLOC_SIZE,
+                             sizeof(maxAllocSize), &maxAllocSize, NULL);
+    error |= clGetDeviceInfo(device, CL_DEVICE_GLOBAL_MEM_SIZE, sizeof(memSize),
+                             &memSize, NULL);
+    test_error(error, "Unable to get max image 1D size from device");
+
+    if (memSize > (cl_ulong)SIZE_MAX)
+    {
+        memSize = (cl_ulong)SIZE_MAX;
+        maxAllocSize = (cl_ulong)SIZE_MAX;
+    }
+
+    if (gTestSmallImages)
+    {
+        for (imageInfo.width = 1; imageInfo.width < 13; imageInfo.width++)
+        {
+            imageInfo.rowPitch = imageInfo.width * pixelSize;
+            if (gDebugTrace) log_info("   at size %d\n", (int)imageInfo.width);
+
+            int ret = test_get_1Dimage_buffer_info_single(
+                context, queue, &imageInfo, seed, flags);
+            if (ret) return -1;
+        }
+    }
+    else if (gTestMaxImages)
+    {
+        // Try a specific set of maximum sizes
+        size_t numbeOfSizes;
+        size_t sizes[100][3];
+
+        get_max_sizes(&numbeOfSizes, 100, sizes, maxWidth, 1, 1, 1,
+                      maxAllocSize, memSize, CL_MEM_OBJECT_IMAGE1D_BUFFER,
+                      imageInfo.format);
+
+        for (size_t idx = 0; idx < numbeOfSizes; idx++)
+        {
+            imageInfo.width = sizes[idx][0];
+            imageInfo.rowPitch = imageInfo.width * pixelSize;
+
+            log_info("Testing %d\n", (int)sizes[idx][0]);
+            if (gDebugTrace)
+                log_info("   at max size %d\n", (int)sizes[idx][0]);
+            if (test_get_1Dimage_buffer_info_single(context, queue, &imageInfo,
+                                                    seed, flags))
+                return -1;
+        }
+    }
+    else
+    {
+        for (int i = 0; i < NUM_IMAGE_ITERATIONS; i++)
+        {
+            cl_ulong size;
+            // Loop until we get a size that a) will fit in the max alloc size
+            // and b) that an allocation of that image, the result array, plus
+            // offset arrays, will fit in the global ram space
+            do
+            {
+                imageInfo.width =
+                    (size_t)random_log_in_range(16, (int)maxWidth / 32, seed);
```

Comment:
Integer overflow here: `maxWidth` is truncated from `size_t` to `int`.
It will cause problems:
`(int)maxWidth` may be a negative integer. As a result, the function `random_log_in_range` will have a second parameter which is negative so it will yield a negative result integer. Then this negative result integer will be converted to `size_t` and `imageInfo.width` will be a very large integer. Finally, the test will fail when trying to allocating memory of a very big size calculated from  `imageInfo.width`.

@rjodinchr Could you please fix the issue?

Reply:
`random_log_in_range` is used more than 200 times in the CTS, with this same pattern many times. It feels like a proper fix would involve changing that function and making sure everything is still working as intended. I'm sorry to say I will  not have time for that in the short term.